common/src/main/java/whocraft/tardis_refined/common/blockentity/device/AstralManipulatorBlockEntity.java

```diff
@@ -0,0 +1,261 @@
+import net.minecraft.world.phys.AABB;
+import org.jetbrains.annotations.Nullable;
+import org.joml.Vector3f;
+import whocraft.tardis_refined.common.crafting.ManipulatorCrafting;
+import whocraft.tardis_refined.common.items.ScrewdriverItem;
+import whocraft.tardis_refined.common.items.ScrewdriverMode;
+import whocraft.tardis_refined.constants.NbtConstants;
+import whocraft.tardis_refined.patterns.ConsolePatterns;
+import whocraft.tardis_refined.registry.BlockEntityRegistry;
+import whocraft.tardis_refined.registry.BlockRegistry;
+import whocraft.tardis_refined.registry.SoundRegistry;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class AstralManipulatorBlockEntity extends BlockEntity {
+
+    // Constants
+    public static final String MAN_POINT_A = "man_point_a";
+    public static final String MAN_POINT_B = "man_point_b";
+    public static final String SHOULD_DISPLAY = "should_display";
+
+
+    private boolean shouldDisplay = false;
+    private BlockPos pointABlockPos;
+    private BlockPos pointBBlockPos;
+
+    public AstralManipulatorBlockEntity(BlockPos blockPos, BlockState blockState) {
+        super(BlockEntityRegistry.ASTRAL_MANIPULATOR.get(), blockPos, blockState);
+    }
+
+    @Override
+    protected void saveAdditional(CompoundTag compoundTag) {
+        super.saveAdditional(compoundTag);
+
+        compoundTag.putBoolean(SHOULD_DISPLAY, this.shouldDisplay);
+
+        if (this.pointABlockPos != null) {
+            compoundTag.put(MAN_POINT_A, NbtUtils.writeBlockPos(this.pointABlockPos));
+        }
+
+        if (this.pointBBlockPos != null) {
+            compoundTag.put(MAN_POINT_B, NbtUtils.writeBlockPos(this.pointBBlockPos));
+
+        }
+    }
+
+    public BlockPos getPointABlockPos() {
+        return this.pointABlockPos;
+    }
+
+    public BlockPos getPointBBlockPos() {
+        return this.pointBBlockPos;
+    }
+
+    public void setShouldDisplay(boolean display) {
+        this.shouldDisplay = display;
+        this.pointABlockPos = this.getBlockPos();
+        this.pointBBlockPos = this.getBlockPos();
+        sendUpdates();
+    }
+
+    public boolean shouldDisplay() {
+
+        return shouldDisplay;
+    }
+
+    public void clearDisplay() {
+        this.shouldDisplay = false;
+        this.pointABlockPos = this.getBlockPos();
+        this.pointBBlockPos = this.getBlockPos();
+        sendUpdates();
+    }
+
+    @Override
+    public void load(CompoundTag tag) {
+
+        if (tag.contains(MAN_POINT_A)) {
+            this.pointABlockPos = NbtUtils.readBlockPos(tag.getCompound(MAN_POINT_A));
+        }
+        if (tag.contains(MAN_POINT_B)) {
+            this.pointBBlockPos = NbtUtils.readBlockPos(tag.getCompound(MAN_POINT_B));
+        }
+
+        if (tag.contains(SHOULD_DISPLAY)) {
+            this.shouldDisplay = tag.getBoolean(SHOULD_DISPLAY);
+        }
+
+        super.load(tag);
+    }
+
+
+    public void OnRightClick(ItemStack itemStack) {
```

Comment:
You appear to be doing onRightClick stuff in the tile and not the block, please move it to the block class as it makes more sense

Also, please watch your naming conventions, OnRightClick vs onRightClick

Reply:
Is there a particular reason it cannot be in the entity as it is the data storage for the process anyhow?

Reply:
There's no reason for it not to be no, but if you look at this way; do you right click a BlockEntity?